src/rgw/rgw_op.cc

```diff
@@ -901,6 +901,17 @@ void rgw_build_iam_environment(rgw::sal::Store* store,
     s->env.emplace("aws:UserAgent", i->second);
   }
 
+  i = m.find("QUERY_STRING"); {
```

Comment:
Let's put a check here as well, if query string itself is empty, we should not move forward.

Reply:
yes,  but the very next line we are checking version id is present or not, if query string empty this won't pass.
  i = m.find("QUERY_STRING"); {
    std::size_t pos = (i->second).find("versionId");
    if( pos != string::npos) {

still we need one more empty check ?

Reply:
Just extended optimization. As discussed, we can skip it as of now.